### PR TITLE
added tutorial: speaking urls for flex object entity pages

### PR DESCRIPTION
**Pull request by bitstarr**

### CODE REVIEW

File: pages/10.cookbook/06.tutorials/02.speaking-urls-for-flex-objects/docs.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+        $debugger->addMessage("Page {$route} already exists, page cannot be added", 'error');
+        return;
+    }
+
+    // grab the single page corpus for later rerouting
+    $page = $pages->find( $fx_page );
+
+    // get the flex object to ensure it exists
+    $collection = $this->grav['flex']->getCollection( $page->header()->fx_directory );
+    $object = $collection->filterBy( ['slug'=> $key] )->first();
```
Comment (bitstarr) [inline pages/10.cookbook/06.tutorials/02.speaking-urls-for-flex-objects/docs.md:148]: those lines could be reduced to
`$object = $this->grav['flex']->getObject( $key, $page->header()->fx_directory, 'slug' );`
but i did not get it to work (returned null), so feel free to improve

**Comment by mahagr** [comment]
There's no need for any of this as you can make the key different from the key used to save the item.

```yaml
config:
  data:
    storage:
      options:
        key: slug
```

The closest example is `user-accounts` type (and yes, there's a way to display it in the frontend, too).

**Comment by bitstarr** [comment]
Thanks for the feedback. As the docs a bit thin about such 'special' needs, can you provide some more info about how to improve this topic in frontend?
You can also find me in the grav discord.

**Comment by mahagr** [comment]
Remind me, right now I'm a bit busy. :) You'll find me from discord, too, though I usually do not participate to the discussions.

**Comment by Sogl** [comment]
@bitstarr did you manage to implement it?

**Comment by bitstarr** [comment]
@Sogl Sorry for the late answer. Indeed I (with the help of the discord server) found a better way.
I implemented it in a plugin, that provides a flex object based event database/list.
https://github.com/bitstarr/grap-plugin-eventlist#pages
https://github.com/bitstarr/grap-plugin-eventlist/blob/main/eventlist.php#L92

**Pull request closed without merging**